tests/fips/libtool/liboauth.pm

```diff
@@ -0,0 +1,102 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: liboauth test for FIPS integration test
+#
+# Description: liboauth is a collection of c functions implementing the http://oauth.net API.
+#              liboauth provides functions to escape and encode stings according to OAuth specifications
+#              and offers high-level functionality built on top to sign requests or verify signatures
+#              using either NSS or OpenSSL for calculating the hash/signatures.
+#
+# Maintainer: Ben Chou <bchou@suse.com>
+# Tags: poo#81260, tc#1767540
+
+use base "consoletest";
+use testapi;
+use utils;
+use utils "zypper_call";
+use strict;
+use warnings;
+use registration qw(add_suseconnect_product cleanup_registration register_product);
+
+sub run {
+    select_console 'root-console';
+
+    zypper_output();
+
+    # Since we use the functional group qcow2 which is regitered to proxyscc
+    # The Source RPM packages are all located in SCC source pool repository
+    # Step 1. SUSEConnect -d deregister from proxyscc
+    # Step 2. SUSEConnect -r <CODE> to SCC
+    # Step 3. SUSEConnect -p module to add DEV tool and Desktop App modules
+
+    # De-register the system from proxyscc due to source RPM package does not mirror to proxyscc
+    cleanup_registration();
+    zypper_output();
+
+    # Register to official SCC(https://scc.suse.com) to get source RPM packages
+    register_product();
+    zypper_output();
+
+    # Add Desktop_Applications_Module_15_SP1_x86_64
+    add_suseconnect_product("sle-module-desktop-applications");
+
+    # Add modules Devlopment Tool Modules
+    add_suseconnect_product("sle-module-development-tools");
+
+    zypper_output();
+
+    # enable source repositories to get latest source packages
+    assert_script_run 'for r in `zypper lr|awk \'/Source-Pool/ {print $5}\'`;do zypper mr -e --refresh $r;done';
+    zypper_output();
+
+    # Install the liboauth source package
+    zypper_call('in -t srcpackage liboauth');
+
+    # libopenssl-devel is required to build liboauth
+    zypper_call('in libopenssl-devel');
+    assert_script_run 'rpm -qi liboauth0';
+
+    # rpm-build is from Development Tools Module 15 SP3 repo
+    # Need to register and Install to Devleopment Tool module first
+    zypper_call('in rpm-build');
+
+    # libcurl-devel & libtool need to be installed due to liboauth dependancy
+    # zypper in libcurl-devel ( Development files for the curl library)
+    # zypper in libtool (A Tool to Build Shared Libraries)
+    zypper_call('in libcurl-devel libtool');
```

Comment:
why not combine these pkgs into a single command?

Reply:
okay, you're right, it is better to combine them in a single cocmmane.

Reply:
DONE